dandi/girder.py

```diff
@@ -290,7 +292,16 @@ def _traverse_asset_girder(self, g, parent_path=None, recursive=True):
             # item should be the one we care about
             pass
         elif a["type"] == "item":
-            file_recs = list(self.listFile(g["_id"]))
+            while True:
+                try:
+                    file_recs = list(self.listFile(g["_id"]))
+                except requests.ConnectionError:
+                    lgr.debug(
+                        "Connection error while listing files; waiting before trying again"
+                    )
+                    time.sleep(0.1)
```

Comment:
this can lead to hanging forever (e.g. if got offline?) and without any feedback to the user ... not sure if desired, and wouldn't be better just to try multiple times until reaching some time out (a minute or so).

Eventually (or right away?) I think we better adopt helpers such as try_multiple* from https://github.com/datalad/datalad/blob/master/datalad/utils.py#L1965 -- we have already a good number of similar points in the code